test/unit/bitmap_test.c

```diff
@@ -0,0 +1,115 @@
+#include <runtime.h>
+#include <stdlib.h>
+#define EXIT_FAILURE 1
+#define EXIT_SUCCESS 0
+
+#define WHOLE_RANGE irange(0, infinity)
+
+
+/**
+ *  Tests allocation of bitmap using allocate_bitmap 
+ *  function.
+ */
+bitmap test_alloc(heap h) {
+    bitmap b = allocate_bitmap(h, h, infinity);
+    bitmap_foreach_set(b, i) {
+        if (i) {
+            msg_err("!!! allocation failed for bitmap\n");
+            return NULL; 
+        }
+    }
+    return b;
+}
+
+/**
+ *  Tests cloning of bitmap using bitmap_clone 
+ *  function.
+ */  
+boolean test_clone(bitmap b) { 
+    for (int i = 0; i < 20; i++) 
+        bitmap_set(b, rand(), 1);
+    bitmap b_cpy = bitmap_clone(b);
+    bitmap_foreach_set(b, j) {
+        if (bitmap_get(b, j) != bitmap_get(b_cpy, j)) {
+            msg_err("!!! cloning failed for bitmap\n");
+            return false;
+        }
+    }
+    return true;
```

Comment:
I updated the bit checking in both clone and copy in my latest commit. Does my implementation test both set and cleared bits?

Reply:
No, it doesn't test cleared bits, because the bitmap_foreach_set loop iterates on set bits only. Also the check using bitmap_base() is unnecessarily complicated, I would replace it with just `if (!bitmap_get(b_cpy, j))`.
To test cleared bits, you could add a new bitmap_foreach_set loop that iterates on `b_cpy` instead of `b`.